fargate/rehydrate/rehydrator.go

```diff
@@ -35,16 +32,13 @@ func (r *Rehydrator) Copy(ctx context.Context, src Source, dest Destination) err
 
 		_, err := r.S3.CopyObject(ctx, &params)
 		if err != nil {
-			log.Printf("error processing simple copy for %s", src.GetName())
-			return err
+			return fmt.Errorf("error processing simple copy for %s: %w", src.GetName(), err)
```

Comment:
This is just meant to format this error and return it and possibly let something further up the chain deal with logging the error right?  Same for like 41 I take it?

Reply:
Yes exactly. I want to avoid accidentally logging the error multiple times, so it's just returning the error, and as you say, letting the code that ultimately handles the error further up the chain log it. Specifically, the `worker` function in `fargate/rehydrate/main.go` passes it into a `WorkerResult` which gets logged by the loop at the end of the `main` function there. 

Ultimately the error or success will be logged to a DynamoDB table and a message put on an SQS queue so that an email can be sent to the user with the result.